python/social_bot/tasks.py

```diff
@@ -422,8 +427,12 @@ def _move_goal_impl(self, goal, agent_loc, agent_dir):
                    random.random() * range - range / 2, 0)
 
             self._initial_dist = np.linalg.norm(loc - agent_loc)
-            if self._initial_dist > self._success_distance_thresh:
+            if self._initial_dist > self._success_distance_thresh and (
+                attempts > 10000 or
+                (abs(loc[0]) < 5 and abs(loc[1]) < 5)  # within walls
```

Comment:
There are a few places in the code where 5 is used. Should make it a variable.

Reply:
Done